infra/src/main/kotlin/fi/fta/geoviite/infra/projektivelho/PVIntegrationService.kt

```diff
@@ -46,38 +45,34 @@ class PVIntegrationService @Autowired constructor(
         lockDao.runWithLock(DatabaseLock.PROJEKTIVELHO, databaseLockDuration) { op() }
     }
 
-    @Scheduled(
-        initialDelayString="\${geoviite.projektivelho.search-poll.initial}",
-        fixedRateString="\${geoviite.projektivelho.search-poll.rate}",
-    )
+    @Scheduled(initialDelayString="\${geoviite.projektivelho.search-poll.initial}", fixedRateString="\${geoviite.projektivelho.search-poll.rate}")
     fun search(): PVApiSearchStatus? = runIntegration {
         logger.info("Poll to launch new search")
-        val latest = pvDao.fetchLatestDocument()
-        val startTime = latest?.second ?: Instant.now().minusSeconds(SECONDS_IN_A_YEAR)
-        pvClient.postXmlFileSearch(startTime, latest?.first).also { status ->
-            val validUntil = status.startTime.plusSeconds(status.validFor)
-            pvDao.insertFetchInfo(status.searchId, validUntil)
+        val currentSearch = pvDao.fetchLatestActiveSearch()
+        if (currentSearch == null) {
```

Comment:
Muutin tämän niin että ei käynnistetä uutta hakua jos edellinen on vielä käynnissä. Normaalilla konffillahan tätä ei pitäisi tapahtua yhdellä instanssilla muutenkaan, koska tämä pollaus tapahtuu paljon myöhemmin kuin käsittely-poll, mutta kahdella instanssilla mitään takuita ei ole. + kehitysympäristössä on kätevää konffia tämä tapahtumaan heti käynnistyksestä.

Reply:
Tämä on ollut tahtotilana aiemminkin, joten se on ollut bugi että näin ei ole tehty aiemmin. Hyvä muutos siis 👏